Source/WTF/wtf/cocoa/FileSystemCocoa.mm

```diff
@@ -239,6 +243,20 @@ bool setExcludedFromBackup(const String& path, bool excluded)
     return true;
 }
 
+bool markPurgeable(const String& path)
+{
+    CString fileSystemPath = fileSystemRepresentation(path);
+    if (fileSystemPath.isNull())
+        return false;
+
+#if HAVE(APFS_CACHEDELETE_PURGEABLE)
+    uint64_t flags = APFS_MARK_PURGEABLE | APFS_PURGEABLE_DATA_TYPE | APFS_PURGEABLE_MARK_CHILDREN;
+    return fsctl(fileSystemPath.data(), APFSIOC_MARK_PURGEABLE, &flags, 0) != -1;
```

Comment:
We should probably check `== 0`. I believe this function returns an error code so the value will differ depending on the error.

Reply:
I thought fsctl returns -1 for error but existing usage seems your suggestion is right; so I will change.